static/app/components/replays/videoReplayer.tsx

```diff
@@ -297,7 +301,12 @@ export class VideoReplayer {
 
     // No more segments
     if (nextIndex >= this._attachments.length) {
-      this.stopReplay();
+      // If we're at the end of a segment, but there's a gap
+      // at the end, force the replay to play until the end duration
+      // rather than stopping right away.
+      this._timer.addNotificationAtTime(this._duration, () => {
+        this.stopReplay();
+      });
```

Comment:
Is there a chance of a race condition here where by the time this notification is added, the timer is already passed duration and will continue on?

Reply:
this one relates to the case where the replay stops early because we've hit a gap/the end of the segment -- the old code here was just `this.stopReplay()`. i think we might be okay here because we don't want to stop early, but rather keep playing until the end, which means we haven't hit `duration` yet

Reply:
I'm thinking about the case where we do not have a gap, and we want to stop the replay:

- segment ends -> handleSegmentEnd is called
- no next segment so this code block is called
- add notification for duration (e.g. 20)
- due to external factors (data, hardware, etc), the current time is > 20
- ?? does the callback get fired in this case?

Reply:
ohhh i see what you're saying! let me add another check that stops the replay if the time is over

Reply:
something like

```
    // No more segments
    if (nextIndex >= this._attachments.length) {
      if (this.getCurrentTime() < this._durationMs) {
        // If we're at the end of a segment, but there's a gap
        // at the end, force the replay to play until the end duration
        // rather than stopping right away.
        this._timer.addNotificationAtTime(this._durationMs, () => {
          this.stopReplay();
        });
        return;
      }
      this.stopReplay();
    }
```